Add documentation for enhanced tab bar preview (#411)
eclipse-theia/theia-website: src/docs/enhanced_tab_bar_preview.md

---
title: Enhanced Tab Bar Preview
---

# Enhanced Tab Bar Preview

By default, Theia shows the value of a widget's `caption` property when users hover above a widget's tab (see also [widgets](/docs/widgets/)).
In certain use cases, especially with custom editors, this information may however not be sufficient to give users sufficient overview about a widget's content before activating the tab.

Therefore, Theia optionally provides an enhanced tab bar preview for widgets of the main or bottom area of Theia; that is, for horizontal tab bars.
This enhanced preview offers more room for information for widgets to show when hovering above a widget's tab.
Moreover, it can easily be styled according to the needs of a tool provider adopting Theia.

> ## Enabling the Enhanced Tab Bar Preview

Reviewer: ```suggestion
## Enabling the enhanced tab bar preview
```

The enhanced tab bar preview is disabled by default and needs to be explicitly enabled with a via the preference `window.tabbar.enhancedPreview`.

<img src="/enhanced-preview-setting.png" alt="A screenshot of the window.tabbar.enhancedPreview setting in Theia" style="max-width: 525px">

After enabling the enhanced tab bar preview, users will see the following on hovering over a widget's tab:

<img src="/enhanced-preview.png" alt="A screenshot of the enhanced preview in Theia" style="max-width: 525px">

> ## Specify caption and label for a widget

Reviewer: ```suggestion
## Specifying the contents of the preview
```

For custom products, the enhanced tab bar preview can be customized and styled according to the requirements of the respective product.
Below we discuss the most relevant customization options.

### Styling of the outer element

The styling of the hover box can easily be done with CSS.
If all hovers should be styled (including ones that are unaffected by the `window.tabbar.enhancedPreview` setting) the `.theia-hover` CSS class can be adjusted.
To only change the styling of the enhanced tab bar previews, use the CSS selector `.theia-hover.extended-tab-preview` instead.
By default, the enhanced tab bar preview for horizontal tab bars has rounded corners. This is defined by adding a `border-radius` to the `.theia-hover.extended-tab-preview` CSS class:

```css
.theia-hover.extended-tab-preview {
    border-radius: 10px;
}
```

### Styling of the content element

To style the elements inside of the preview, class names are assigned to the different components of the preview.
Those are:

- `.theia-horizontal-tabBar-hover-div` (for the outer box)
- `.theia-horizontal-tabBar-hover-title` (for the title)
- `.theia-horizontal-tabBar-hover-caption` (for the caption).

> If, for example, the preview should be a fixed size. This can bew achieved by adding a `width` to the `.theia-horizontal-tabBar-hover-div` and a `max-width` to the other two rules.

Reviewer: ```suggestion
If, for example, the preview should have a fixed size, this can be achieved by specifying a `width` for the `.theia-horizontal-tabBar-hover-div` and a `max-width` to the other two rules.
```
